src/cli.rs

```diff
@@ -294,6 +294,12 @@ pub struct Opt {
     /// set this in per-repository git config (.git/config)
     pub default_language: Option<String>,
 
+    /// Detect whether or not the terminal is dark or light by querying for its colors.
+    ///
+    /// Ignored if either `--dark` or `--light` is specified.
+    #[arg(long = "detect-dark-light", value_enum, default_value_t = DetectDarkLight::default())]
+    pub detect_dark_light: DetectDarkLight,
```

Comment:
Nice, I like the name. My first thought would have been something-something-terminal-something, but this increases coherency by tying it to the existing `light` and `dark` options.